src/mappings/token.ts

```diff
@@ -21,14 +17,25 @@ export function fetchTokenDetails(event: ethereum.Event): Token | null {
   //check if token details are already saved
   let token = Token.load(event.address.toHex());
   if (!token) {
-    //if token details are not available
-    //create a new token
+    // If token details are not available, create a new token
     token = new Token(event.address.toHex());
 
+    //set the address field
+    token.address = event.address;
+
     //set some default values
     token.name = "N/A";
     token.symbol = "N/A";
     token.decimals = 0;
+    token.eventCount = ZERO; // Initialize eventCount
+    token.burnEventCount = ZERO; // Initialize eventCount
```

Comment:
Necessary?

Reply:
removed. keep tranferCount and totalTransferred. those can be useful.